pydantic/experimental/pipeline.py

```diff
@@ -22,6 +23,11 @@
 
 from pydantic._internal._internal_dataclass import slots_true as _slots_true
 
+if sys.version_info < (3, 10):
+    EllipsisType = type(Ellipsis)
+else:
+    from types import EllipsisType
```

Comment:
What happens with Python < 3.10 then? What sort of type checker errors pop up?

Reply:
I've confirmed we're now back to this working:

```py
validate_as(str).transform(lambda x: x.strip()) # _Pipeline[Any, str]
```

Reply:
Right but what errors do users get when using the API (if any)? We need to check pyright in strict mode.

Reply:
```py
from pydantic.experimental.pipeline import validate_as

int_pipeline = validate_as(str).validate_as(int)# _Pipeline[Any, int]
str_pipeline = validate_as(str).transform(lambda x: x.strip()) # _Pipeline[Any, str]

reveal_type(int_pipeline)
#> pydantic/test.py:9:13 - information: Type of "int_pipeline" is "_Pipeline[Any, int]"
reveal_type(str_pipeline)
#> pydantic/test.py:10:13 - information: Type of "str_pipeline" is "_Pipeline[Any, str]"
```

Reply:
what about `validate_as(…)`?

Reply:
Well we can't do static type checking there, but it does work at runtime:

```py
from typing import Annotated

from pydantic.experimental.pipeline import validate_as
from pydantic import TypeAdapter

ta = TypeAdapter(Annotated[int, validate_as(str).validate_as(...)])
print(ta.validate_python('1'))
#> 1
```

Reply:
Indeed everything seems to work now!

```python
(
    validate_as(...)
    .transform(lambda x: x.strip())  # no error, no type checking (Any)
    .transform(str.strip)
    .transform(lambda x: x.strip())  # type checked as a string
    .validate_as(int)
    .transform(lambda x: x.strip())  # error, type checking as an int
)
```